src/game.zig

```diff
@@ -684,16 +684,55 @@ pub fn update(deltaTime: f64) void { // MARK: update()
 				}
 			}
 			if(KeyBoard.key("backward").pressed) {
-				movementSpeed = @max(movementSpeed, 4);
-				movementDir += forward*@as(Vec3d, @splat(-4));
+				if(KeyBoard.key("sprint").pressed) {
+					if(Player.isGhost.load(.monotonic)) {
+						movementSpeed = @max(movementSpeed, 128);
+						movementDir += forward*@as(Vec3d, @splat(-128));
+					} else if(Player.isFlying.load(.monotonic)) {
+						movementSpeed = @max(movementSpeed, 32);
+						movementDir += forward*@as(Vec3d, @splat(-32));
+					} else {
+						movementSpeed = @max(movementSpeed, 4);
+						movementDir += forward*@as(Vec3d, @splat(-4));
+					}
+				} else {
+					movementSpeed = @max(movementSpeed, 4);
+					movementDir += forward*@as(Vec3d, @splat(-4));
+				}
 			}
 			if(KeyBoard.key("left").pressed) {
-				movementSpeed = @max(movementSpeed, 4);
-				movementDir += right*@as(Vec3d, @splat(4));
+				if(KeyBoard.key("sprint").pressed) {
+					if(Player.isGhost.load(.monotonic)) {
+						movementSpeed = @max(movementSpeed, 128);
+						movementDir += right*@as(Vec3d, @splat(128));
+					} else if(Player.isFlying.load(.monotonic)) {
+						movementSpeed = @max(movementSpeed, 32);
+						movementDir += right*@as(Vec3d, @splat(32));
+					} else {
+						movementSpeed = @max(movementSpeed, 4);
+						movementDir += right*@as(Vec3d, @splat(4));
+					}
+				} else {
+					movementSpeed = @max(movementSpeed, 4);
+					movementDir += right*@as(Vec3d, @splat(4));
+				}
 			}
 			if(KeyBoard.key("right").pressed) {
-				movementSpeed = @max(movementSpeed, 4);
-				movementDir += right*@as(Vec3d, @splat(-4));
+				if(KeyBoard.key("sprint").pressed) {
+					if(Player.isGhost.load(.monotonic)) {
+						movementSpeed = @max(movementSpeed, 128);
+						movementDir += right*@as(Vec3d, @splat(-128));
+					} else if(Player.isFlying.load(.monotonic)) {
+						movementSpeed = @max(movementSpeed, 32);
+						movementDir += right*@as(Vec3d, @splat(-32));
+					} else {
+						movementSpeed = @max(movementSpeed, 4);
+						movementDir += right*@as(Vec3d, @splat(-4));
+					}
+				} else {
+					movementSpeed = @max(movementSpeed, 4);
+					movementDir += right*@as(Vec3d, @splat(-4));
+				}
```

Comment:
I think there is a more elegant solution if you use `movementSpeed`.

Reply:
So, something like what I'm about to commit? It does cause the player to be able to sprint in the directions on ground.

Reply:
KInd of. But you should readd a special case when on the ground.

Reply:
I did that, that also added diagonal sprinting like in Minecraft.

Reply:
Honestly, I think this diagonal sprinting feels wrong. Can't you just go back to the previous logic when not flying?

Reply:
I just tried that and I cant seem to get it to work.

Reply:
Any progress on this or have you given up?